Shared/Samples/Animate images with image overlay/AnimateImagesWithImageOverlayView.swift

```diff
@@ -0,0 +1,190 @@
+    /// The view model for the sample.
+    class Model: ObservableObject {
+        /// A scene with a dark gray base and centered on southern California.
+        let scene: ArcGIS.Scene = {
+            // Creates a scene and sets an initial viewpoint.
+            let scene = Scene(basemapStyle: .arcGISDarkGrayBase)
+            let point = Point(x: -116.621, y: 24.7773, z: 856977.0, spatialReference: .wgs84)
+            let camera = Camera(location: point, heading: 353.994, pitch: 48.5495, roll: 0)
+            scene.initialViewpoint = Viewpoint(boundingGeometry: point, camera: camera)
+            
+            // Add base surface from elevation service.
+            let elevationSource = ArcGISTiledElevationSource(url: .worldElevationService)
+            let surface = Surface()
+            surface.addElevationSource(elevationSource)
+            scene.baseSurface = surface
+            return scene
+        }()
+        
+        /// A timer to synchronize image overlay animation to the refresh rate of the display.
+        var displayLink: CADisplayLink!
+        
+        /// An iterator to hold and loop through the overlay images.
+        private var imagesIterator: CircularIterator<UIImage> = {
+            // Get the URLs to images added to the project's folder reference.
+            let imageURLs = Bundle.main.urls(forResourcesWithExtension: "png", subdirectory: "PacificSouthWest2") ?? []
+            let images = imageURLs
+                .sorted { $0.lastPathComponent < $1.lastPathComponent }
+                .map { UIImage(contentsOfFile: $0.path)! }
+            return CircularIterator(elements: images)
+        }()
+        
+        /// A formatter to format percentage strings.
+        let percentageFormatter: NumberFormatter = {
+            let formatter = NumberFormatter()
+            formatter.numberStyle = .percent
+            formatter.multiplier = 100
+            return formatter
+        }()
+        
+        /// The frame rate speed of the image overlay
+        @Published var fpsSpeed = 30
+        
+        /// The image overlay to show image frames.
+        @Published var imageOverlay: ImageOverlay = {
+            let imageOverlay = ImageOverlay()
+            imageOverlay.opacity = 0.5
+            return imageOverlay
+        }()
+        
+        init() {
+            // Create new display link.
+            let newDisplayLink = CADisplayLink(target: self, selector: #selector(setImageFrame))
+            
+            // Inherit the frame rate from existing display link, or set to default 60 fps.
+            newDisplayLink.preferredFramesPerSecond = displayLink?.preferredFramesPerSecond ?? 60
+            newDisplayLink.isPaused = true
+            
+            // Add to main thread common mode run loop, so it is not effected by UI events.
+            newDisplayLink.add(to: .main, forMode: .common)
+            displayLink = newDisplayLink
+        }
```

Comment:
In the old sample this code is in `makeDisplayLink` because after the `displayLink` is invalidated, we need to create a new instance when the main view reappears.

If you still stick to the `onAppear/onDisappear` view lifecycle, then I recommend to move back to the old approach.

Indeed, in this sample there is no way to let the main view disappear, but our users may have such usecase, and in that case the display link won't work with code, when the view appears for the 2nd time.